### ISSUE TITLE
Question about BETYdb load

### ISSUE THREAD

**Issue by serbinsh** [PecanProject/pecan#425]
Hello again @robkooper and @dlebauer, @Viskari and I were finally able to get everything operating properly (I think) after overhauling the gcc/gfortran dn openmpi libs then re-compiling everything using a consistent gcc lib.  As such, I have made an attempt to load bety locally (see below).  Based on your experiences, does this look correct?  My one concern are the:

ERROR:  must be owner of extension plpgsql
ERROR:  must be owner of extension postgis

Reports.  Otherwise, it seems that all the data has been loaded.

My load command was as follows:

sudo -u postgres CREATE=YES MYSITE=3 scripts/load.bety.sh

Did I miss anything?  Do I need postgres or bety to take ownership of the plpgsql and postgis extensions?  Do I need to run this command from the postgres user rather than as sudo? 

Thanks!
S

```
[sserbin@test pecan]$ sudo -u postgres CREATE=YES MYSITE=3 scripts/load.bety.sh
 bety      
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 4641k  100 4641k    0     0  4488k      0  0:00:01  0:00:01 --:--:-- 8691k
Loading schema                    : NOTICE:  drop cascades to 4 other objects
DETAIL:  drop cascades to extension postgis
drop cascades to extension postgis_topology
drop cascades to extension fuzzystrmatch
drop cascades to extension postgis_tiger_geocoder
CREATE EXTENSION
GRANT
ERROR:  must be owner of extension plpgsql
ERROR:  must be owner of extension postgis
CREATED SCHEMA
Loading  schema_migrations         : ADDED 35
Cleaning citations                 : DEL 0
Loading  citations                 : ADD 779
Fixing   citations                 : SEQ 3000000001
Cleaning counties                  : DEL 0
Loading  counties                  : ADD 3399
Fixing   counties                  : SEQ 3000000001
Cleaning covariates                : DEL 0
Loading  covariates                : ADD 10475
Fixing   covariates                : SEQ 3000000001
Cleaning cultivars                 : DEL 0
Loading  cultivars                 : ADD 517
Fixing   cultivars                 : SEQ 3000000001
Cleaning dbfiles                   : DEL 0
Loading  dbfiles                   : ADD 814
Fixing   dbfiles                   : SEQ 3000000001
Cleaning ensembles                 : DEL 0
Loading  ensembles                 : ADD 2
Fixing   ensembles                 : SEQ 3000000001
Cleaning entities                  : DEL 0
Loading  entities                  : ADD 516
Fixing   entities                  : SEQ 3000000001
Cleaning formats                   : DEL 0
Loading  formats                   : ADD 34
Fixing   formats                   : SEQ 3000000001
Cleaning inputs                    : DEL 0
Loading  inputs                    : ADD 248
Fixing   inputs                    : SEQ 3000000001
Cleaning likelihoods               : DEL 0
Loading  likelihoods               : ADD 0
Fixing   likelihoods               : SEQ 3000000001
Cleaning location_yields           : DEL 0
Loading  location_yields           : ADD 23081
Fixing   location_yields           : SEQ 3000000001
Cleaning machines                  : DEL 0
Loading  machines                  : ADD 13
Fixing   machines                  : SEQ 3000000001
Cleaning managements               : DEL 0
Loading  managements               : ADD 4011
Fixing   managements               : SEQ 3000000001
Cleaning methods                   : DEL 0
Loading  methods                   : ADD 37
Fixing   methods                   : SEQ 3000000001
Cleaning mimetypes                 : DEL 0
Loading  mimetypes                 : ADD 417
Fixing   mimetypes                 : SEQ 3000000001
Cleaning models                    : DEL 0
Loading  models                    : ADD 5
Fixing   models                    : SEQ 3000000001
Cleaning modeltypes                : DEL 0
Loading  modeltypes                : ADD 3
Fixing   modeltypes                : SEQ 3000000001
Cleaning modeltypes_formats        : DEL 0
Loading  modeltypes_formats        : ADD 9
Fixing   modeltypes_formats        : SEQ 3000000001
Cleaning pfts                      : DEL 0
Loading  pfts                      : ADD 84
Fixing   pfts                      : SEQ 3000000001
Cleaning posterior_samples         : DEL 0
Loading  posterior_samples         : ADD 0
Fixing   posterior_samples         : SEQ 3000000001
Cleaning posteriors                : DEL 0
Loading  posteriors                : ADD 4
Fixing   posteriors                : SEQ 3000000001
Cleaning priors                    : DEL 0
Loading  priors                    : ADD 216
Fixing   priors                    : SEQ 3000000001
Cleaning runs                      : DEL 0
Loading  runs                      : ADD 23
Fixing   runs                      : SEQ 3000000001
Cleaning sessions                  : DEL 0
Loading  sessions                  : ADD 9425
Fixing   sessions                  : SEQ 3000000001
Cleaning sites                     : DEL 0
Loading  sites                     : ADD 1071
Fixing   sites                     : SEQ 3000000001
Cleaning species                   : DEL 0
Loading  species                   : ADD 49708
Fixing   species                   : SEQ 3000000001
Cleaning treatments                : DEL 0
Loading  treatments                : ADD 2481
Fixing   treatments                : SEQ 3000000001
Cleaning variables                 : DEL 0
Loading  variables                 : ADD 488
Fixing   variables                 : SEQ 3000000001
Cleaning workflows                 : DEL 0
Loading  workflows                 : ADD 8
Fixing   workflows                 : SEQ 3000000001
Cleaning traits                    : DEL 0
Loading  traits                    : ADD 13050
Fixing   traits                    : SEQ 3000000001
Cleaning yields                    : DEL 0
Loading  yields                    : ADD 6153
Fixing   yields                    : SEQ 3000000001
Cleaning users                     : DEL 0
Loading  users                     : ADD 165
Fixing   users                     : SEQ 3000000001
Cleaning citations_sites           : DEL 0
Loading  citations_sites           : ADD 1247
Cleaning citations_treatments      : DEL 0
Loading  citations_treatments      : ADD 1949
Cleaning formats_variables         : DEL 0
Loading  formats_variables         : ADD 61
Cleaning inputs_runs               : DEL 0
Loading  inputs_runs               : ADD 0
Cleaning inputs_variables          : DEL 0
Loading  inputs_variables          : ADD 31
Cleaning managements_treatments    : DEL 0
Loading  managements_treatments    : ADD 9063
Cleaning pfts_priors               : DEL 0
Loading  pfts_priors               : ADD 1475
Cleaning pfts_species              : DEL 0
Loading  pfts_species              : ADD 200511
Cleaning posteriors_ensembles      : DEL 0
Loading  posteriors_ensembles      : ADD 0
```

**Comment by serbinsh** [comment]
Evidence that we have populated the DB

```
bety=# SELECT * FROM models;
 id | model_name | revision | parent_id |         created_at         |         updated_at         | modeltype_id 
----+------------+----------+-----------+----------------------------+----------------------------+--------------
  3 | ED2.2      | 46       |           | 2011-05-17 16:45:44        | 2011-10-12 20:42:22        |            1
  4 | ED2.1      | 42       |           | 2011-05-17 16:47:41        | 2011-10-12 20:42:12        |            1
  5 | ED2.2      | 80       |           | 2012-04-05 13:26:07        | 2012-04-05 13:38:47        |            1
  7 | BioCro     |          |           | 2013-09-18 18:34:50        | 2013-09-18 14:36:08        |            2
 10 | SIPNET     | unk      |           | 2014-04-19 02:00:30.708947 | 2014-04-19 02:00:30.708947 |            3
(5 rows)
```

**Comment by mdietze** [comment]
I think your MYSITE is supposed to be 2 for Brookhaven, not 3 (which is Purdue).

Can you pull from other sites as well (e.g. REMOTESITE=1 for BU) ? Are you dumping to an accessible location that the rest of us can try pulling from?

**Comment by serbinsh** [comment]
Hi Mike

Correct, I used the wrong MYSITE.  I will need to reload with the correct value

As for a dump location, I may need to discuss that.  Currently we have the database being stored on our data partition under /data/database.  We could open the up (i think) if needed.  However, I think I need a power lessen or a discussion on what the setup should be to allow for remote access.  Of course, that can't happen until our server has a dedicated hole in our firewall.  As soon as that happens we can give it a test.

For now, I am just trying to run some local tests to see if everything is running properly.  @mdietze or @robkooper if I re-run the load.bety script and use the CREATE=YES option then it should wipe out my previous load and reload it from scratch, correct?  I want to make sure we correct the MYSITE mixup.

**Comment by serbinsh** [comment]
I corrected location issue (I think) by running:

sudo -u postgres CREATE=YES MYSITE=2 scripts/load.bety.sh

**Comment by mdietze** [comment]
Shawn, the dump is different from where the database is stored and from opening up remote access to the database itself -- running dump.bety.sh should create a tar.gz file that you can post anywhere for us to download (http, ftp, etc), which could potentially even include rsync'ing it to another server that's not firewalled. Both ebi-forecast and psql-pecan run the dump on a cron job (for me the same cron job is running the upload of ebi-forecast).

More generally, how will you run the web utils if the firewall's blocking http and/or https?

**Comment by serbinsh** [comment]
Hi Mike

Got it.  We actually discussed this before but I forgot the details.  We will dump the database on an open share, as you mentioned an http or FTP dir, that will allow for rsync or automatic FTP retrieval.  

As for web utils, etc  we will be working with ITD to open up access to our server.  That is allow remote requests.  Currently the only work within the BNL network.  That should be happening within the next couple of weeks.

So to summarize, we will create a location that can be accessed via scp/rsync/ftp and will be opening up remote access to our sever.

**Comment by serbinsh** [comment]
Fix for DB load.  Add paths to postgres user.
